### PR TITLE
Lock quorum

### PR DESCRIPTION
**Pull request by mmedenjak**
Adding quorum to lock actions. We are reusing the existing quorum support which means that this does not guarantee correctness, just best-effort detection of quorum. This will be emphasised in the  documentation and in the TDD.

The lock configuration element/class was added since lock didn't yet have configuration.

Note to keep in mind when reviewing - adding quorum for locks affects locking methods on map and multimap as well.

### CODE REVIEW

File: hazelcast/src/main/resources/hazelcast-config-3.8.xsd [hunk 1]
```diff
@@ -1172,6 +1173,23 @@
             </xs:simpleType>
         </xs:attribute>
     </xs:complexType>
+
+    <xs:complexType name="lock">
```
Comment (vbekiaris) [inline hazelcast/src/main/resources/hazelcast-config-3.8.xsd]: Consider adding support for serializing the new elements in `ConfigXmlGenerator` (but feel free to add it later with a separate PR, it's not crucial).

Comment (mmedenjak) [inline hazelcast/src/main/resources/hazelcast-config-3.8.xsd]: Right. There's the spring config as well, I will create a new PR for that.

Comment (mmedenjak) [inline hazelcast/src/main/resources/hazelcast-config-3.8.xsd]: I've added lock config to config xml generator and spring config. Also fixed queue and ringbuffer config in xml generator.

**Comment by eminn** [comment]
Apart from the comments @vbekiaris mentioned, looks good to be merged 👍

**Comment by mmedenjak** [comment]
@eminn Thanks for the review!

**Pull request merged**